Updated Changelog for MSAL 4.46.2 (#3630)
AzureAD/microsoft-authentication-library-for-dotnet: CHANGELOG.md

4.46.2
==========

### New Features

> - WAM Authentication Library Should Support .NET 4.5 for Back Compatibility. See [#3539](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3539)

Reviewer: ```suggestion
- WAM Authentication Library now explicitly supports .NET 4.6.2. See [#3539](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3539)
```

Reviewer: (and yes, I know that it actually targets 4.6.1, but 4.6.1 is out of support. 4.6.2 is the only one still supported)

> - Log exceptions which are not converted to MsalExceptions to telemetry. See [#3547](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3547)

Reviewer: Not a feature, I'd say this is a bug fix.

> - WAM Authentication Library Should Support .NET 4.5 for Back Compatibility. See [#3539](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3539)
> - Log exceptions which are not converted to MsalExceptions to telemetry. See [#3547](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3547)
> 
> ### Bug Fixes
> - Authenticator Factory has already been started. See [#3604](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3604)
> - MSAL.NativeInterop only targets .net fwk See [#3612](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3612)

Reviewer 2: nit:

```suggestion
- Added .NET 4.5 support for new MSAL WAM preview. See [#3539](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3539)
- Logging additional exceptions to telemetry. See [#3547](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3547)

### Bug Fixes
- Fixed 'Authenticator Factory has already been started` exception in new MSAL WAM preview. See [#3604](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3604)
- Added back missing .NET Standard 2.0 target to MSAL.NativeInterop package. See [#3612](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3612)
```

> 

Reviewer 2: ```suggestion
- [Resilience] Changed to an improved implementation of HTTP client factory on .NET Framework to improve resiliency (for ex. by reducing the amount of request timeouts). See [#3546](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3546)

```
